### PR TITLE
feat(suggested-assignee): include release committers in suggested assignees

### PR DESCRIPTION
**Pull request by snigdhas**
Include all release committers for a group in the suggested assignees.

This PR builds off of #37697 to surface release committers on both the Issue Stream and the Issue Details page. These changes are a V1, and will be need to be optimized before releasing to a larger audience. [Tech Spec](https://www.notion.so/sentry/Mini-Spec-Release-Committers-as-Suggested-Assignees-d5a0723f73e147349ac65e8aa941f300)

Fixes WOR-2069 and WOR-2094.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_event_committers.py [hunk 1]
```diff
@@ -2,14 +2,18 @@
...
 from sentry.testutils.factories import DEFAULT_EVENT_DATA
 from sentry.testutils.helpers.datetime import before_now, iso_format
+from sentry.testutils.helpers.features import apply_feature_flag_on_cls
 from sentry.utils.samples import load_data
 
-# TODO(dcramer): These tests rely too much on implicit fixtures
-
 
+# TODO(dcramer): These tests rely too much on implicit fixtures
+@apply_feature_flag_on_cls("organizations:release-committer-assignees")
```
Comment (barkbarkimashark) [inline tests/sentry/api/endpoints/test_event_committers.py:16]: lol nice ...

File: src/sentry/models/groupowner.py [hunk 2]
```diff
@@ -1,24 +1,34 @@
...
     GroupOwnerType.SUSPECT_COMMIT: "suspectCommit",
     GroupOwnerType.OWNERSHIP_RULE: "ownershipRule",
+    GroupOwnerType.RELEASE_COMMIT: "releaseCommit",
 }
 
 
+class OwnersSerialized(TypedDict):
+    type: GroupOwnerType
+    author: str
+    commits: models.DateTimeField
```
Comment (barkbarkimashark) [inline src/sentry/models/groupowner.py:29]: field should be `date_added`?

File: src/sentry/models/groupowner.py [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
 class GroupOwnerType(Enum):
     SUSPECT_COMMIT = 0
     OWNERSHIP_RULE = 1
+    RELEASE_COMMIT = 2
 
 
 GROUP_OWNER_TYPE = {
     GroupOwnerType.SUSPECT_COMMIT: "suspectCommit",
     GroupOwnerType.OWNERSHIP_RULE: "ownershipRule",
+    GroupOwnerType.RELEASE_COMMIT: "releaseCommit",
```
Comment (barkbarkimashark) [inline src/sentry/models/groupowner.py:22]: Not sure how I feel about prematurely adding these here. Might give off the impression we;re storing these in the db when we're not.

File: src/sentry/models/groupowner.py (hunk 2 above)
Comment (snigdhas) [reply src/sentry/models/groupowner.py]: oops, fixed

File: src/sentry/models/groupowner.py (hunk 3 above)
Comment (snigdhas) [reply src/sentry/models/groupowner.py]: good point, removed

File: src/sentry/api/endpoints/event_file_committers.py [hunk 4]
```diff
@@ -36,17 +37,11 @@ def get(self, request: Request, project, event_id) -> Response:
...
-        #         'commits': serialize(frame['commits'])
-        #     } for frame in annotated_frames
-        # ]
-
         data = {
             "committers": committers,
-            # 'annotatedFrames': serialized_annotated_frames
         }
+
+        if features.has("organizations:release-committer-assignees", project.organization):
```
Comment (scttcper) [inline src/sentry/api/endpoints/event_file_committers.py:44]:
```suggestion
        if features.has("organizations:release-committer-assignees", project.organization, actor=request.user):
```

can we pass user to support roll out to teams/users

File: src/sentry/models/groupowner.py [hunk 5]
```diff
@@ -63,17 +80,82 @@ def owner(self):
...
-            }
+            OwnersSerialized(
+                type=GROUP_OWNER_TYPE[GroupOwnerType(go.type)],
+                owner=go.owner().get_actor_identifier(),
+                date_added=go.date_added,
+            ),
         )
 
+    org = group_list[0].project.organization if len(group_list) > 0 else None
+    if org and features.has("organizations:release-committer-assignees", org):
```
Comment (scttcper) [inline src/sentry/models/groupowner.py:97]: if we could somehow pass the user here as well

File: src/sentry/models/groupowner.py [hunk 6]
```diff
@@ -19,6 +24,18 @@ class GroupOwnerType(Enum):
 }
 
 
+class OwnersSerialized(TypedDict):
+    type: GroupOwnerType
+    owner: str
+    date_added: models.DateTimeField
```
Comment (scttcper) [inline src/sentry/models/groupowner.py:30]:
```suggestion
    dateAdded: models.DateTimeField
```
should be camelcase

File: src/sentry/models/groupowner.py [hunk 7]
```diff
@@ -19,6 +24,18 @@ class GroupOwnerType(Enum):
 }
 
 
+class OwnersSerialized(TypedDict):
+    type: GroupOwnerType
+    owner: str
+    dateAdded: models.DateTimeField
```
Comment (scttcper) [inline src/sentry/models/groupowner.py:30]: nevermind keep this as date_added. my bad, didn't realize we had already shipped date_added and you're adding types for it

File: src/sentry/models/groupowner.py (hunk 5 above)
Comment (snigdhas) [reply src/sentry/models/groupowner.py:97]: Done

**Review by scttcper — APPROVED** [review]

**Pull request merged**